### PR TITLE
P(X < oo) and P(X > oo) for True, False probability

### LINKED ISSUE #10052: P(X < oo ) for any Continuous Distribution raises AttributeError
```
>>> from sympy import *
>>> from sympy.stats import P, Exponential
>>> X = Exponential('X', 3)
>>> P(X < oo)

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-5-c320871e22f8> in <module>()
----> 1 P(X < oo)

/home/gxyd/Public/sympy/sympy/stats/rv.py in probability(condition, given_condition, numsamples, evaluate, **kwargs)
    617 
    618     # Otherwise pass work off to the ProbabilitySpace
--> 619     result = pspace(condition).probability(condition, **kwargs)
    620     if evaluate and hasattr(result, 'doit'):
    621         return result.doit()

AttributeError: 'NoneType' object has no attribute 'probability'
```

### PR DESCRIPTION
**Pull request by gxyd**
fixes #10052

### CODE REVIEW

**Comment by jksuom** [comment]
This looks good.

There are other (nonsensical) cases where ``pspace(condition)`` returns ``None`` and ``AttributeError`` results, for example ``P(1)``. Perhaps these should also be caught and something more informative be returned (maybe ``nan``).
```

File: sympy/stats/rv.py [hunk 1]
```diff
@@ -608,6 +609,13 @@ def probability(condition, given_condition=None, numsamples=None,
     5/12
     """
 
+    if given_condition is S.false:
```
Comment (asmeurer) [inline sympy/stats/rv.py]: Better to use `==` than `is` here.

Comment (gxyd) [inline sympy/stats/rv.py]: Your concern seems right to me.

**Comment by gxyd** [comment]
> Perhaps these should also be caught and something more informative be returned (maybe nan).

Yes, i have added `ValueError`, but since `condition` is expected to be a `Relational`, so i think returning `ValueError` looks better to me instead of `nan`.

**Comment by jksuom** [comment]
It is a good idea to catch any ``condition`` that is not a ``Relational``.
In addition, I think it would be advisable to make sure that ``pspace(condition)`` does not return ``None``. That can happen even when ``condition`` is a ``Relational`` if it does not contain random variables. For example, ``P(x < 1)`` for an ordinary symbol ``x``.

**Comment by gxyd** [comment]
> It is a good idea to catch any condition that is not a Relational.

It can even take non-relationals like: `And(X + Y < 3, Y < 2) `. That's why
the current tests fail. I think i should use some, `.has(Relational)`
property  for that may be.

**Comment by gxyd** [comment]
Ex. P(And(X + Y < 2, X > 1)) where `X` and `Y` are both `RandomSymbol`.

**Comment by jksuom** [comment]
Yes, there are ``BooleanFunction``s like ``And`` that should be accepted. Perhaps everything that derives from ``Boolean`` should be allowed. That would simplify the test.

**Comment by gxyd** [comment]
I too think it is better for `given(condition)` to raise an Error instead of returning `None`. It is now ready for review.

**Comment by gxyd** [comment]
I have addressed all the comments, though I `rebased` the PR, which led to the last two comments you mentioned to be removed(i have addressed them as well), since they were commented on PR-commits. Sorry for that.

**Comment by jksuom** [comment]
No problem. The comments were only for temporary use.

**Comment by asmeurer** [comment]
PyPy seems to be having some problems. It looks unrelated to this PR.

**Comment by jksuom** [comment]
Just one job. I can restart it.

**Comment by jksuom** [comment]
I think this is ready. I'll merge in 12 hours.

**Comment by gxyd** [comment]
Thanks @jksuom @asmeurer

**Pull request merged**